src/server/pachyderm_test.go

```diff
@@ -10676,14 +10701,23 @@ func TestDatumSetCache(t *testing.T) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 	go func() {
-		ticker := time.NewTicker(time.Minute)
+		ticker := time.NewTimer(50 * time.Second)
 		defer ticker.Stop()
 		for {
 			select {
 			case <-ctx.Done():
 				return
 			case <-ticker.C:
+				// ticker is too smart and will adjust to always go at 1 minute, even if etcd is slow.
+				// So, we pause the ticker while the pod restart occurs so we know how long the pod is executing for.
 				tu.DeletePod(t, "etcd", ns)
+				tu.WaitForFirstPodReady(t, "etcd", ns)
```

Comment:
This seems like the only place where we call DeletePod and there is a lot of action at a distance here (it doesn't merely delete the pod, it also waits for a replacement to be scheduled, but not ready).  I think it would be better to do this logic somewhere in this test file and do something more reliable than DeletePod; scale the etcd StatefulSet to 0, wait for there to be no matching etcd pods, scale etcd to 1, wait for matching pod to be ready.  Overall DeletePod chooses an unreliable and abstraction-breaking algorithm for the sake of being general, but this is the only place that calls it, so it doesn't seem like a good tradeoff to me.